Some Python3 compatibility fixes.

cultcom opened the pull request:
I tried debops within a Python 3 virtual environment and encountered some exceptions. Maybe there is more to be fixed but this could be a starting point.

Commit:
Some Python3 compatibility fixes.

Commit:
Use BytesIO instead of StringIO.

Commit:
Use StringIO with unicode string.

Just using BytesIO breaks Python3 support. With explicitly defining
"DEFAULTS" as unicode string it works with versions 2 and 3.
With this commit "debops-update" runs in both environments without
exceptions so far.

Commit:
Fix python3 print and PEP8 warning.

drybjed reviewed (changes requested):
Awesome, thanks. While you're at it, can you update the `debops-defaults` script to look in the correct location for the roles? It should look in the monorepo directory, but it expects `debops-playbooks`.

Commit:
Make debops-default PEP8 compliant.

Commit:
More Python3 + PEP8 fixes.

"debops" runs in both environments now (Python 2 and 3) but still throws
a template exception error 'dict object' has no attribute 'iteritems'
while running with v3.

Commit:
Merge remote-tracking branch 'origin/master' into python3

Commit:
Replace "iteritems()" by "items()" for jinja.

items() is ineffecient for Python 2 but compatible with Pyhton 3.

ypid reviewed (changes requested):
Thanks very much for working on this! Two minor details attached.

Commit:
Fixed nsswitch.fact.j2 syntax.

Commit:
A first suggestion for a coding policy extension.
